### ISSUE TITLE
Firestore emulator: False negative warning about affecting production when running tests

### ISSUE THREAD

**Issue by noelmansour** [firebase/firebase-tools#1530]
### [REQUIRED] Environment info

**firebase-tools:** master (dad143c42445056014f6f48cc9dfa13156e3c186)

**Platform:** macOS

### [REQUIRED] Test case

Discovered while testing out the fix for: #1451.

Please note I'm not familiar with javascript nor typescript so it's very likely there is something wrong with my setup. Happy to be told that's the case.

src/index.ts:
```typescript
import * as admin from 'firebase-admin';

admin.initializeApp();

export * from './debug'
```

src/debug.ts:
```typescript
import * as admin from 'firebase-admin';
import * as functions from 'firebase-functions';
import Timestamp = admin.firestore.Timestamp;

export const foo = functions.https.onRequest(async (req, resp) => {
    const firestore = admin.firestore();
    await firestore.doc('/users/foo/').set({
        time: Timestamp.now(),
        enabled: true,
    });
    resp.sendStatus(200);
});
```

test/index.test.ts:
```typescript
import * as assert from 'assert';

import('../src/index');

describe('Cloud Functions', () => {

    describe('foo', () => {
        it('bar', () => {
            assert.strictEqual(true, true);
        });
    });
});
```

test/mocha.opts:
```
--ui mocha-typescript
--require ts-node/register test/**/*.ts
--recursive test
```

package.json:
```json
{
  "name": "functions",
  "scripts": {
    "lint": "tslint --project tsconfig.json",
    "build": "tsc",
    "serve": "npm run build && firebase serve --only functions",
    "shell": "npm run build && firebase functions:shell",
    "emulators": "npm run build && firebase emulators:start",
    "start": "npm run shell",
    "deploy": "firebase deploy --only functions",
    "logs": "firebase functions:log",
    "format": "prettier --write **/*.ts",
    "pretest": "tsc",
    "test": "mocha"
  },
  "engines": {
    "node": "8"
  },
  "main": "lib/index.js",
  "dependencies": {
    "firebase-admin": "^8.2.0",
    "firebase-functions": "^3.1.0"
  },
  "devDependencies": {
    "@types/mocha": "^5.2.7",
    "mocha": "^6.1.4",
    "mocha-typescript": "^1.1.17",
    "prettier": "^1.18.2",
    "ts-node": "^8.3.0",
    "tslint": "^5.12.0",
    "typescript": "^3.5.2"
  },
  "private": true
}
```

tsconfig.json:
```json
{
  "compilerOptions": {
    "module": "commonjs",
    "noImplicitReturns": true,
    "noUnusedLocals": true,
    "outDir": "lib",
    "sourceMap": true,
    "strict": true,
    "target": "es2015",
    "typeRoots" : ["./node_modules/@types", "./typings"]
  },
  "compileOnSave": true,
  "include": [
    "src"
  ]
}
```
### [REQUIRED] Steps to reproduce

Run `npm run build`. Then run `firebase emulators:exec "npm run test"`.

### [REQUIRED] Expected behavior

No warning.

### [REQUIRED] Actual behavior

Shows this warning:
`⚠  The Cloud Firestore emulator is not running, so calls to Firestore will affect production.`

Logs from running with `--debug` [debug.txt](https://github.com/firebase/firebase-tools/files/3412896/debug.txt)

**Comment by samtstern** [comment]
@noelmansour what happens if you remove this line:
```
import Timestamp = admin.firestore.Timestamp;
```

And then just use `Timestamp` fully-qualified below:
```
    await firestore.doc('/users/foo/').set({
        time: admin.firestore.Timestamp.now(),
        enabled: true,
    });
```

**Comment by noelmansour** [comment]
Hm, yeah that fixed it.

Looks like I need to read up a bit more on typescript and javascript imports. :sweat_smile:

**Comment by samtstern** [comment]
@noelmansour I still think this is a bug on our side but I'm glad the fix was simple.

**Comment by samtstern** [comment]
I just got this myself:
```
$ npx firebase emulators:start
i  Starting emulators: ["functions","firestore","database"]
⚠  Your requested "node" version "8" doesn't match your global version "10"
✔  functions: Emulator started at http://localhost:5001
i  firestore: Emulator logging to firestore-debug.log
✔  firestore: Emulator started at http://localhost:8080
i  firestore: For testing set FIRESTORE_EMULATOR_HOST=localhost:8080
i  database: downloading firebase-database-emulator-v4.1.0.jar...
Progress: ==========================================================> (100% of 18MB)
i  database: Removing outdated emulator: firebase-database-emulator-v4.0.0.jar
i  database: Emulator logging to database-debug.log
✔  database: Emulator started at http://localhost:9000
i  functions: Watching "/tmp/tmp.tkVh7jwr4l/functions" for Cloud Functions...
⚠  The Cloud Firestore emulator is not running, so calls to Firestore will affect production.
✔  functions: [http] function helloJSON initialized (http://localhost:5001/fir-dumpster/us-central1/helloJSON).
✔  functions: [http] function info initialized (http://localhost:5001/fir-dumpster/us-central1/info).
✔  functions: [http] function writeToFirestore initialized (http://localhost:5001/fir-dumpster/us-central1/writeToFirestore).
✔  functions: [http] function writeToDatabase initialized (http://localhost:5001/fir-dumpster/us-central1/writeToDatabase).
✔  functions: [firestore] function firestoreReaction initialized.
✔  functions: [database] function databaseReaction initialized.
i  functions: Function authFunction ignored because the emulator for firebaseauth.googleapis.com does not exist or is not running.
```

**Comment by samtstern** [comment]
This should be fixed in `7.2.0`
